### PR TITLE
Update Auto Detection of Person and allow for Unknown to be displayed

### PR DESCRIPTION
**Pull request by chakrabortyr**
## Description
Properly adds an unknown user type, to be assigned when we don't have a person ID.

## Motivation and Context
We need Federated Users with no person ID to be properly associated with an "Unknown" user.

## Tests performed
Yeah

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: html/internal_dashboard/js/admin_panel/SectionExistingUsers.js [hunk 1]
```diff
@@ -1173,6 +1167,7 @@ XDMoD.ExistingUsers = Ext.extend(Ext.Panel, {
                         }
                         else {
                             cmbInstitution.setDisabled(true);
+                            cmbInstitution.initializeWithValue(json.user_information.institution, json.user_information.institution_name);
```
Comment (plessbd) [inline html/internal_dashboard/js/admin_panel/SectionExistingUsers.js]: This seems out of scope for this specific change, I would suggest not doing this at this time...

Comment (smgallo) [reply html/internal_dashboard/js/admin_panel/SectionExistingUsers.js]: Ditto. Don't set the institution until we flesh out the requirements document for new user creation.

File: classes/Authentication/SAML/XDSamlAuthentication.php [hunk 2]
```diff
@@ -90,7 +90,12 @@ public function getXdmodAccount()
         } else {
             $thisUserName = !empty($samlAttrs['username'][0]) ? $samlAttrs['username'][0] : null;  # code...
         }
-        if ($this->_as->isAUthenticated() && !empty($thisUserName)) {
+        if (!isset($samlAttrs["system_username"])) {
+            $thisSystemUserName = $thisUserName;
+        } else {
+            $thisSystemUserName = !empty($samlAttrs['system_username'][0]) ? $samlAttrs['system_username'][0] : null;  # code...
```
Comment (smgallo) [inline classes/Authentication/SAML/XDSamlAuthentication.php]: Remove comment at end of line.
